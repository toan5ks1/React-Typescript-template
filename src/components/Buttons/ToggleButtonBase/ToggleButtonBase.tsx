import { ToggleButton, ToggleButtonProps } from "@mui/material";
import { styled } from "@mui/material/styles";

export interface ToggleButtonBaseProps extends ToggleButtonProps {}

const PREFIX = "ToggleButtonBase";

const classes = {
    selected: `${PREFIX}-selected`,
    root: `${PREFIX}-root`,
};

const StyledToggleButton = styled(ToggleButton)(({ theme }) => ({
    [`&.${classes.root}`]: {
        color: theme.palette.text.secondary,
        padding: "6px 1rem", //design confirm,
    },
    [`&.${classes.selected}`]: {
        color: theme.palette.primary.main,
        backgroundColor: theme.palette.primary.background,
        borderColor: theme.palette.primary.border,
    },
}));

const ToggleButtonBase = (props: ToggleButtonBaseProps) => {
    const { value, children, classes: classOverride, ...rest } = props;

    return (
        <StyledToggleButton
            value={value}
            classes={{
                root: classes.root,
                selected: classes.selected,
                ...classOverride,
            }}
            data-testid="ToggleButtonBase"
            {...rest}
        >
            {children}
        </StyledToggleButton>
    );
};

export default ToggleButtonBase;
