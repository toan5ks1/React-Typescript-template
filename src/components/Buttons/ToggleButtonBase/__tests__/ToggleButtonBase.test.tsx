import ToggleButtonBase, { ToggleButtonBaseProps } from "../ToggleButtonBase";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<ToggleButtonBase />", () => {
    let wrapper: RenderResult;

    const props: ToggleButtonBaseProps = {
        value: "test",
        children: "Test",
    };

    beforeEach(() => {
        wrapper = render(<ToggleButtonBase {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
