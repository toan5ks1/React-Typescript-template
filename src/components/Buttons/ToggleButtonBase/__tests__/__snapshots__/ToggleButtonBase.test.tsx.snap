// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButtonBase /> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-selected {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-0.Mui-selected:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }
}

.emotion-0.ToggleButtonBase-root {
  color: rgba(0, 0, 0, 0.6);
  padding: 6px 1rem;
}

.emotion-0.ToggleButtonBase-selected {
  color: #1976d2;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <button
    class="MuiButtonBase-root ToggleButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard emotion-0"
    data-testid="ToggleButtonBase"
    tabindex="0"
    type="button"
    value="test"
  >
    Test
    <span
      class="MuiTouchRipple-root emotion-1"
    />
  </button>
</div>
`;
